GNUmakefile

```diff
@@ -8,6 +8,8 @@
 default: $(shell test ! -e configure && echo init) $(shell test -e configure && echo core)
 
 -include Makefile
+# order important
+-include maint/make/speed-up.mk
```

Comment:
Let's just put the contents here rather than use a separate file. This code can affect anyone running the regression tests from a checkout, not just maintainers.

Reply:
I don't want to despute location of the make "module", can be elsewhere,
but I'd like to ask for this modularity to be revisited.  Actually, this
was my take-away from libvirt codebase some years back that it prevents
monolithic spaghetti build system when pieces are an overlay addons.
Btw. this is also the approach of `gnulib`, even if not isolated within
single layer, but also spanning `m4` files, hence configure time, etc.

Reply:
And actually, the point of using that location was exactly to help
_anyone_ using the checkout (there's no question whether one is
a maintainer anywhere :-), but not to affect pristine tarball.
Pristine tarball users are still welcome to pass job request
explicitly (while checkout users would get the best guess
estimate effortlessly).

Reply:
The other supporting argument would be that these optional Makefile
extensions can have deliberately relaxed treatment wrt. portability
if we aim to ever restore POSIX make compatibility for the core.

Reply:
Keeping this tentative as well.

Reply:
I think this is good as it is.
We shouldn't second guess what the non-checkout builders actually want.